com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariableBase.cs

```diff
@@ -79,6 +79,10 @@ public void Initialize(NetworkBehaviour networkBehaviour)
         public virtual void SetDirty(bool isDirty)
         {
             m_IsDirty = isDirty;
+            if (m_NetworkBehaviour != null && m_IsDirty)
```

Comment:
This check should also be added to Initialize, as I've seen it's possible for this code to run before Initialize.

Reply:
If this does crash, do you think you could replicate it?  If so, then we might come up with a better order of operations for this to assure that case never happens.  Maybe in the end this is the best fix for the underlying issue you exeperienced, but if it is just masking a larger order of operations issue that could be fixed at a higher level (in the call stack) then it might be better to fix the issue there.

Reply:
ok, figured out the reason, adjusted tests, removed the extra check.